src/logger/LoggerFactory.cpp

```diff
@@ -25,7 +25,7 @@ namespace sql
 {
 namespace mariadb
 {
-  Shared::Logger LoggerFactory::NO_LOGGER= (NO_LOGGER ? NO_LOGGER : Shared::Logger(new NoLogger()));
+  Shared::Logger LoggerFactory::NO_LOGGER= nullptr;
```

Comment:
there is probably no need to assign it to nullptr - I think default constructor will do the same thing, but for cheaper.

Reply:
I've just pushed this line the way I want it(along with one change I needed). I suspect it should do the job with the warning